### PR TITLE
Test: add coverage for genesis consistency

### PR DESCRIPTION
**Pull request by lgalabru**
This PR is testing the predictability of the chainstate process, by ensuring that we're always reproducing the same MARF root hash.

### CODE REVIEW

File: src/chainstate/stacks/db/mod.rs [hunk 1]
```diff
@@ -1819,4 +1820,125 @@ pub mod test {
...
+            marf.get_root_hash_at(&index_block_hash).unwrap()
+        });
+        assert_eq!(
+            format!("{}", genesis_block),
+            "e336d28e3bccdf8e5e77216e0dea26b60292f4577b05ab2d4414776f29a5e9fc"
+        );
+    }
+
+    #[test]
+    #[ignore]
```
Comment (jcnelson) [inline src/chainstate/stacks/db/mod.rs:1884]: Why is this `ignore`?  Does it take a long time to run?

Comment (lgalabru) [reply src/chainstate/stacks/db/mod.rs:1884]: Yeah I had to mark this test as ignored in https://github.com/blockstack/stacks-blockchain/pull/2188/commits/5638835b9177a84964432a001dc7769d39348f6f after OOM'ing github's runner

**Review by kantai — APPROVED** [review]
This looks good to me -- just add a note to that test case that it would need to be updated in any PR that modifies the genesis data.

**Review by jcnelson — APPROVED** [review]
LGTM!

**Pull request merged**